### PR TITLE
Select most recent JDK

### PR DESCRIPTION
**Pull request by raboof**
When multiple versions of, say, jdk8 are installed, pick the highest one.

### CODE REVIEW

**Comment by raboof** [comment]
(appveyor error seems to be #4238)

**Review by eed3si9n — APPROVED** [review]
Nice

File: main/src/main/scala/sbt/internal/CrossJava.scala [hunk 1]
```diff
@@ -299,47 +300,78 @@ private[sbt] object CrossJava {
...
+            case dir @ JavaHomeDir(vendor, m, n) =>
+              val v = JavaVersion(nullBlank(m) + n).withVendor(vendor).toString
+              if ((base / dir / "Contents" / "Home").exists) v -> (base / dir / "Contents" / "Home")
+              else v -> (base / dir)
+          }
     }
+
+    val configs = Vector(
+      new JabbaDiscoverConfig,
+      new LinuxDiscoverConfig(file("/usr") / "java"),
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/internal/CrossJava.scala:371]: This adds support for Fedora. Ref https://github.com/akka/akka/issues/25298

**Comment by MasseGuillaume** [comment]
This won't work in nix/nixos, you need to provide the list explicitly.

**Comment by raboof** [comment]
@MasseGuillaume interesting, perhaps we should open a separate ticket about making this feature work with nix/nixos. I'm only superficially familiar with nixos, how would multiple installed JDK versions look from 'inside' a nix environment?

**Comment by MasseGuillaume** [comment]
You create your environment like this:

https://github.com/scala-native/scala-native/blob/master/scripts/scala-native.nix#L10

You just need to be able to inject dependencies with an environment variable.

For example:

```
ORACLE_JDK_8 = pkgs.oraclejdk8; // available in all-packages.nix
```

https://github.com/NixOS/nixpkgs/blob/6b5c9fc197895ad2f3b63b473168a5871670eee3/pkgs/top-level/all-packages.nix#L6720

Then sbt will have `ORACLE_JDK_8` available via `sys.env`

**Comment by MasseGuillaume** [comment]
https://github.com/sbt/sbt/issues/4260

**Pull request merged**